src/cmap/command_monitoring_events.ts

```diff
@@ -161,6 +195,15 @@ export class CommandFailedEvent {
   }
 }
 
+/* @internal */
+function parseServerConnectionID(serverConnectionId: number | Double | bigint | Long): bigint {
+  return Long.isLong(serverConnectionId)
+    ? serverConnectionId.toBigInt()
+    : typeof serverConnectionId === 'bigint' || typeof serverConnectionId === 'number'
+    ? BigInt(serverConnectionId)
+    : BigInt(serverConnectionId.valueOf());
```

Comment:
It's not letting me cast a Double using the above code sample. Will using valueOf() instead lead to an incorrect implementation?

Reply:
Can you elaborate on the issue you are seeing? 

It would not be incorrect, but unnecessary, a branching code path that does not need to exist could be removed so we should avoid adding that kind of debt to this logic.

Reply:
The JS BigInt constructor does not take in BSON Double's as input.
```
var BigInt: BigIntConstructor (value: string | number | bigint | boolean) => bigint
```
Were you thinking of using a different BigInt to support this?

Reply:
- https://tc39.es/ecma262/#sec-bigint-constructor-number-value
- https://tc39.es/ecma262/#sec-toprimitive
- https://tc39.es/ecma262/#sec-ordinarytoprimitive

That is really unfortunate that the TS library type declares such restrictive types for the input when it does not reflect the reality of what the runtime supports. 

This is an ideal place for using a [`ts-expect-error`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments) comment to work around the limitation so we can get notified if this type is ever fixed.